packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts

```diff
@@ -110,13 +110,19 @@ export class HostedPluginProcess implements ServerPluginRunner {
 
     }
 
+    readonly exclusion = ['HOSTED_PLUGIN*'];
     private fork(options: IPCConnectionOptions): cp.ChildProcess {
 
         // create env and add PATH to it so any executable from root process is available
-        const env = createIpcEnv();
-        env.PATH = process.env.PATH;
-        // add HOME to env since some plug-ins need to read files from user's home dir
-        env.HOME = process.env.HOME;
+        const env = createIpcEnv({ env: process.env });
+        for (const excl of this.exclusion) {
```

Comment:
just a FYI, it looks like many lines to delete something from an object. Looks like we could reduce code with map or filter